packages/asset-ownership-service/src/services/substream.ts

```diff
@@ -0,0 +1,566 @@
+        substreamPackage: substream,
+        outputModule: MODULE,
+        productionMode: PRODUCTION,
+        startBlockNum: cursor ? undefined : currentBlock,
+        startCursor: cursor,
+      });
+
+      console.log(
+        `Substream: Streaming from ${
+          cursor ? `cursor ${cursor}` : `block ${currentBlock}`
+        }`
+      );
+
+      attemptCount = 0;
+      isConnecting = false;
+
+      for await (const response of streamBlocks(transport, request)) {
+        if (currentAbortController.signal.aborted) {
+          return;
+        }
+
+        if (shouldRestart) {
+          shouldRestart = false;
+          currentAbortController.abort();
+          return;
+        }
+
+        const message = response.message;
+
+        if (message.case === "fatalError") {
+          console.error("Substream error:", message.value);
+          throw new Error("Received fatal error from substream");
+        }
+
+        if (message.case === "blockScopedData") {
+          const output = unpackMapOutput(response, registry);
+          const cursor = message.value.cursor;
+          const blockHeight =
+            message.value.finalBlockHeight?.toString() || "unknown";
+
+          const hasTransactions =
+            output !== undefined &&
+            !isEmptyMessage(output) &&
+            (output as any).transactions.length > 0;
+
+          const t = await database.transaction({
+            isolationLevel: Transaction.ISOLATION_LEVELS.READ_COMMITTED,
+          });
+
+          try {
+            if (hasTransactions) {
+              const outputTransactions = (output as any)
+                .transactions as IOutputTransaction[];
+
+              const filteredTransactions = outputTransactions.filter((tx) =>
+                tx.meta.logMessages.some((log) =>
+                  RELEVENT_INSTRUCTIONS.some((instr) => log.includes(instr))
+                )
+              );
+
+              if (filteredTransactions.length > 0) {
+                const lookupTableCache: Record<
+                  string,
+                  AddressLookupTableAccount
+                > = {};
+
+                await Promise.all(
+                  filteredTransactions.map(async (transactionInfo) => {
+                    const transaction = await convertSubstreamTransaction(
+                      transactionInfo
+                    );
+
+                    if (!transaction) return;
+                    const { message } = transaction;
+                    const { addressTableLookups } = message;
+                    const addressLookupTableAccounts: AddressLookupTableAccount[] =
+                      [];
+
+                    for (const addressTableLookup of addressTableLookups) {
+                      const key = addressTableLookup.accountKey.toBase58();
+                      if (!lookupTableCache[key]) {
+                        const lookupTableResult =
+                          await provider.connection?.getAddressLookupTable(
```

Comment:
Doesn't this call already happen in convertSubstreamTransaction? And it's not cached there. Though the account fetch cache kind of does it